### DISCUSSION TITLE
How to show loading state when calling API in effects?

### DISCUSSION THREAD

**Discussion by WenLonG12345** [rematch/rematch#971] [Q&A]
Hi, I want to show loading state when calling API in effects.
I had tried `@rematch/loading` but the loading state keep returns `false`.

Here is my implementation.

```
const store = init({
  models: {
    profileModel,
    authModel,
    connectionModel,
    cardModel,
  },
  plugins:[
    persistPlugin(persistConfig),
    loadingPlugin()
  ],
  redux: {
    rootReducers: {
      RESET_ALL_STATE: () => undefined
    }
  }
});

export default store;
```

```
const mapState = ({ loading }) => ({
  loading: loading.effects.cardModel.getSelfCart,  
});

const CartPage = ({ token, dispatch, loading }) => {

    useEffect(() => {
       dispatch({
         type: 'cardModel/getSelfCart',  <-- calling API in effects
         }
       })
     }, [])

    console.log('loading',loading) < -- keep return as false
}

export default connect(mapState)(CartPage);
```

**Comment by semoal** [comment]
Hi!

Are you sure your effect `getSelfCart`is async? If you can create a codesandbox or similar I could help much better.

**Reply by WenLonG12345** [reply]
Hi sure,

https://codesandbox.io/embed/white-moon-pr3zmk?fontsize=14&hidenavigation=1&theme=dark

**Reply by semoal** [reply]
Looks like the type notation is not working correctly. I'll take a look asap.
Meanwhile you can use the Rematch kind of dispatch.

Copy and paste this and check that it will work fine:

```js
import { useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";

const App = () => {
  const dispatch = useDispatch();
  const isLoading = useSelector(state => state.loading.effects.cartModel.getSelfCart)
  useEffect(() => {
    dispatch.cartModel.getSelfCart();
  }, []);

  return (
    <div className="App">
      <h2>Is loading? {isLoading.toString()}</h2>
    </div>
  );

export default App;
```

Replaced the `connect` HOC, by the Hooks connection because is the recommended approach by the Rematch and the Redux team.

**Reply by WenLonG12345** [reply]
Hi @semoal , thanks for your suggestion, I notice that `dispatch` from `connect` HOC is not fixing the issue. but `useDispatch()` fixes it. What is the different between `dispatch from connect` and `dispatch from useDispatch()`?

**Reply by semoal** [reply]
Will look more deeply, probably react-redux changed something internally! Thanks for opening the discussion

**Reply by WenLonG12345** [reply]
Should I report this in rematch's issue? it seems confusing for developer with pure redux background.
